### PR TITLE
Setup gem & setup automatic gem publishing

### PR DESCRIPTION
**Pull request by janzenisaac**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  publish:
+    if: contains(github.ref, 'refs/tags/v')
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Release Gem
+        uses: discourse/publish-rubygems-action@main
```
Comment (cvx) [inline .github/workflows/ci.yml]: Could you use v2 instead? See this for an upgrade example: https://github.com/discourse/onebox/pull/484/files

File: discourse_zendesk_api.gemspec [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  #   'bug_tracker_uri' => 'https://github.com/zendesk/zendesk_api_client_rb/issues',
+  #   'changelog_uri' => "https://github.com/zendesk/zendesk_api_client_rb/blob/v#{s.version}/CHANGELOG.md",
+  #   'documentation_uri' => "https://www.rubydoc.info/gems/zendesk_api/#{s.version}",
+  #   'source_code_uri' => "https://github.com/zendesk/zendesk_api_client_rb/tree/v#{s.version}",
+  #   'wiki_uri' => 'https://github.com/zendesk/zendesk_api_client_rb/wiki'
+  # }
+
+  s.files = Dir.glob('{lib,util}/**/*') << 'LICENSE'
+
+  s.required_ruby_version     = ">= 2.3"
```
Comment (cvx) [inline discourse_zendesk_api.gemspec]:
```suggestion
  s.required_ruby_version     = ">= 2.5"
```

**Review by cvx — CHANGES_REQUESTED** [review]
If we're running specs in our workflow, the `main.yml` one can be removed.

**Comment by janzenisaac** [comment]
@CvX I cant re-request you 🙃 the button won't work, even on refresh.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```
Comment (cvx) [inline .github/workflows/ci.yml]: One last thing: we can drop that, since we're on `main` already 😃 

```suggestion
```

**Review by cvx — APPROVED** [review]

**Pull request merged**